src/invidious/yt_backend/connection_pool.cr

```diff
@@ -88,3 +88,18 @@ def make_client(url : URI, region = nil, &block)
     client.close
   end
 end
+
+# Fetches a HTTP pool for the specified subdomain of ytimg.com
+#
+# Creates a new one when the specified pool for the subdomain does not exist
+def get_ytimg_pool(subdomain)
+  if pool = YTIMG_POOLS[subdomain]?
+    return pool
+  else
+    LOGGER.info("ytimg_pool: Creating a new HTTP pool for \"https://#{subdomain}.ytimg.com\"")
+    pool = YoutubeConnectionPool.new(URI.parse("https://#{subdomain}.ytimg.com"), capacity: CONFIG.pool_size)
```

Comment:
Does the pool not automatically clean up connections?

Reply:
I don't think so? It seems that released connections are pushed to the `idle` list, but I'm not sure what causes them to be closed:
https://github.com/crystal-lang/crystal-db/blob/3eaac85a5d4b7bee565b55dcb584e84e29fc5567/src/db/pool.cr#L151-L185

Reply:
`YoutubeConnectionPool` sets the `max_pool_size` and `max_idle_pool_size` to the value of `capacity`. So meaning the pool will start out with 0 clients, and create more up to `capacity`, but then it'll keep all of them around forever.

There is also not currently a way to set a timeout on idle connections. I.e. that would expire idle connections after some period of time. Is some discussion related to this in https://github.com/crystal-lang/crystal-db/issues/47.

In the meantime it might be worth allowing to customize the `max_idle_pool_size` vs always assuming it should be `CONFIG.pool_size` big. E.g. have a higher `max_pool_size` to handle bursts, but then keep `max_idle_pool_size` set to something smaller for normal traffic. `1` might be a good starting point, could run some benchmarks to figure out a good value.